packages/cli/src/load-nodes-and-credentials.ts

```diff
@@ -260,6 +264,34 @@ export class LoadNodesAndCredentials {
 		return loader;
 	}
 
+	/**
+	 * This creates all AI Agent tools by duplicating the node descriptions for
+	 * all nodes that are marked as `usableAsTool`. It basically modifies the
+	 * description. The actual wrapping happens in the langchain code for getting
+	 * the connected tools.
+	 */
+	createAiTools() {
+		const usableNodes: Array<INodeTypeBaseDescription | INodeTypeDescription> =
+			this.types.nodes.filter((nodetype) => nodetype.usableAsTool === true);
+
+		for (const usableNode of usableNodes) {
+			const description: INodeTypeBaseDescription | INodeTypeDescription =
+				structuredClone(usableNode);
+			const wrapped = NodeHelpers.convertNodeToAiTool({ description }).description;
+
+			this.types.nodes.push(wrapped);
+			this.known.nodes[wrapped.name] = this.known.nodes[usableNode.name];
```

Comment:
Is it ok that we'll have two records pointing to the same object? Does it make sense to use structuredClone here as well?

Reply:
I don't think it would matter here as we just generate JSON object with the node definition but it's definitely a valid point. Updated — thanks!